dimitriosv/week2/index.html

```diff
@@ -0,0 +1,316 @@
+                  allMessages: allMessages,
+                  totalMessages: this.state.totalMessages+1
+
+            });
+        },
+        deleteMessageParent: function(messageId) {
+            var allMessages=this.state.allMessages;
+
+            var newArray = allMessages.filter(function(obj) {
+                return messageId!=(obj.messageId) 
+            });
+                       
+            this.setState({
+                  allMessages: newArray
+            });
+             
+        },
+        archiveMessageParent: function(messageId) {
+            var allMessages=this.state.allMessages;
+            allMessages.forEach(function(obj) {
+                if (obj.messageId === messageId) {
+                   if (obj.IsArchived) {
+                        obj.IsArchived=false;
+                   } else {
+                         obj.IsArchived=true;
+                   }
+                }
+            });
+ 
+            this.setState({
+                  allMessages: allMessages
+            });
+        },
+        handleChange: function(evt) {
+            this.setState({
+              inputListName: evt.target.value
+          });
+        },
+        render: function() {
+            return  <div className="container">
+                        <span>Add new list with name: </span>
+                        <input onChange={this.handleChange} type="text"/>
+                        <button type="button" onClick={this.createList}>Add</button>
+                        <div className="llistcont">
+                            {this.state.lists.map(this.eachList)}
+                        </div>
+                    </div>
+        },
+    });
+
+    var List = React.createClass({
+        getInitialState: function() {
+            return {
+                    inputMessageName:"",
+                    listId:this.props.listId
+                };
+        },
+        render: function() {
+            return  <div className="llist">
+                        <div className="listname" >{this.props.listName}</div>
+                        <div>
+                            <input onChange={this.handleMessageChange} type="text"/>
+                            <button type="button" onClick={this.addMessageChild}>Submit</button>
+                        </div>
+                         {/*console.log(thats how you comment inside return)*/}
+                         <Messages  deleteMessageParent={this.props.deleteMessageParent}
+                                    archiveMessageParent={this.props.archiveMessageParent}
+                                    addMessageParent={this.props.addMessageParent}
+                                    messages={this.filterMessages()} 
+                                    allLists={this.props.allLists}/> 
+                    </div>
+        },
+        addMessageChild: function() {
+            if (this.state.inputMessageName.length==0 || this.state.inputMessageName.length>200) {
```

Comment:
`if (this.state.inputMessageName || ...` instead of `if (this.state.inputMessageName.length==0 || ...` would be enough.